cmd/subctl/reconstruct.go

```diff
@@ -0,0 +1,154 @@
+	controllerClient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var recoverRestConfigProducer = restconfig.NewProducer().WithPrefixedContext("broker").WithDefaultNamespace(constants.OperatorNamespace).
+	WithPrefixedNamespace("broker", constants.DefaultBrokerNamespace)
+
+var (
+	ipsecFile  string
+)
+
+// recoverBrokerInfo represents the reconstruct command.
+var recoverBrokerInfo = &cobra.Command{
+	Use:   "recover-brokerinfo",
+	Short: "Recovers broker-info.subm file from the installed Broker",
+	Run: func(cmd *cobra.Command, args []string) {
+		exit.OnError(recoverBroker(cli.NewReporter()))
+	},
+}
+
+func init() {
+	addReconstructFlags()
+	recoverRestConfigProducer.SetupFlags(recoverBrokerInfo.Flags())
+	rootCmd.AddCommand(recoverBrokerInfo)
+}
+
+func addReconstructFlags() {
+	recoverBrokerInfo.PersistentFlags().StringVar(&ipsecFile, "ipsec-psk-from", "",
+		"import IPsec PSK from existing submariner broker file, like broker-info.subm")
+}
+
+func recoverBroker(status reporter.Interface) error {
+
+	// if --brokerconfig flag provided, get the broker and proceed to get Submariner
+	contextFound, err := recoverRestConfigProducer.RunOnSelectedPrefixedContext("broker",
+		func(brokerCluster *cluster.Info, brokerNamespace string, status reporter.Interface) error {
+			status.Start("Checking if Broker is installed on cluster %s in %s namespace", brokerCluster.Name, brokerNamespace)
+			defer status.End()
+			err := errors.New("error: ")
+			found, err := brokerCluster.HasBroker(brokerNamespace)
+			if err != nil {
+				return err
+			}
+			if found {
+				status.Success("Broker found")
+				status.Start("Checking if Submariner is installed on the same cluster")
+
+				if brokerCluster.Submariner != nil {
+					status.Success("Submariner found")
+					fmt.Printf("getting details")
+					return broker.ReconstructData(brokerCluster, brokerCluster, brokerNamespace, status, ipsecFile)
+				}
+				status.Warning("Submariner is not installed on this cluster. Checking if Submariner is installed on a different cluster")
+				return recoverRestConfigProducer.RunOnSelectedContext(
+					func(submCluster *cluster.Info, namespace string, status reporter.Interface) error {
+						if submCluster.Submariner != nil {
+							status.Success("Submariner is installed on cluster %s", submCluster.Name)
+							fmt.Printf("getting details")
+							return broker.ReconstructData(brokerCluster, submCluster, namespace, status, ipsecFile)
+						}
+
+						return status.Error(err, "submariner is not installed on this cluster. " +
+							"Please specify the cluster where Submariner is installed by using `--kubeconfig` or `--kubecontext` flag")
+					}, status)
+			}
+
+			return status.Error(err, "Broker is not installed on the specified cluster. Please try again with another cluster")
+
+		}, status)
+
+	// if --brokerconfig not provided, search for broker on current context and proceed to get Submariner
+	if !contextFound && err == nil {
```

Comment:
Avoids all the indentation.

```suggestion
	if contextFound || err != nil {
	    return err
	}
```